03_simon_says/simon_says.rb

```diff
@@ -0,0 +1,38 @@
+def shout(str)
+    str.upcase
+  end
+  
+  def echo(str)
+    str
+  end
+  
+  def repeat(str, times=2)
+    temp = str
+    (times-1).times{
+      str = str + " " + temp
+      #str << " " + temp
+      #De ce nu merge?
```

Comment:
Esti sigur ca nu merge?

Reply:
Da, mi-am dat seama si de ce. 
Prin `temp=str` am creeat un pointer temp la acelasi obiect string. Am verificat prin a compara `str.object_id` si `temp.object_id`, sunt egale.
Cand modificam `str` prin `str << " " + temp` modificam in heap obiectul la care pointeaza  `str` si `temp`. `temp` devine noul `str` la urmatoarea iteratie iar asta rezulta intr-un rezultat eronat.
O solutie ar fi declararea `temp =String.new(str)` pentru a creea un obiect now.